src/jit/codegenxarch.cpp

```diff
@@ -4641,6 +4641,15 @@ void CodeGen::genCallInstruction(GenTreePtr node)
     // all virtuals should have been expanded into a control expression
     assert(!call->IsVirtual() || call->gtControlExpr || call->gtCallAddr);
 
+    // Insert a GS check if necessary
+    if (call->IsTailCallViaHelper())
+    {
+        if (compiler->getNeedsGSSecurityCookie())
+        {
+            genEmitGSCookieCheck(true);
```

Comment:
> To be able to use a reg safely in genEmitGSCookieCheck(), we either need to reserve an internal register on the call node or materialize explicitly in IR as GT_GSCHECK node in the right place with register requirements.

Carol and I had discussed the latter and had decided that it was not necessary, but we did not consider the fact that in NGEN/Crossgen scenarios we may need a register. I've updated this PR to introduce a `GT_GS_CHECK` node that is now used to represent all GS cookie checks when using the RyuJIT backend.